### PR TITLE
Bugfix/115 bug in 2 of the detectors transmittedmtofxandyandsubregionhistdetector and reflectedmtofxandyandsubregionhistdetector

### PR DESCRIPTION
**Pull request by hayakawa16**

### CODE REVIEW

**Comment by lmalenfant** [comment]
@hayakawa16 the Sonar Cloud analysis failed because the duplication is 52.5% for this code, I have it set to 20% on our rule but I guess it will go over that number even after we merge (20.8%). This duplication number has been creeping up so even setting it at 20% was not high enough, should I increase it more?

**Comment by hayakawa16** [comment]
Not sure I understand.  Did the duplication increase due to my code changes?  Or is it the same percentage as it was before my changes?

**Comment by hayakawa16** [comment]
If I'm reading the analysis correctly, it is due to the renaming of the file ReflectedMTOfRhoAndSubregionHistDetector.cs.  Not sure why.

**Comment by lmalenfant** [comment]
> Not sure I understand. Did the duplication increase due to my code changes? Or is it the same percentage as it was before my changes?

No it didn't but I think now that the file schanged, it calculates based on "new code" so this is actually both our changes that caused new code to show 52.5%. The previous code coverage percentage was 20.8% so maybe merging this to master will not fail the analysis. I could also raise it to 21% just to pass this build.

File: src/Vts.Test/MonteCarlo/Detectors/DAWLayersDetectorsTests.cs [hunk 1]
```diff
@@ -29,22 +29,22 @@ public class DAWLayersDetectorsTests
...
-        private readonly double _dosimetryDepth = 1.0;
+        private const double LayerThickness = 1.0; // tissue is homogeneous (both layer opt. props same)
+        private const double DosimetryDepth = 1.0;
         private double _factor;
 
         /// <summary>
         /// list of temporary files created by these unit tests
         /// </summary>
-        readonly List<string> listOfTestGeneratedFiles = new List<string>()
+        private readonly List<string> _listOfTestGeneratedFiles = new List<string>
```
[the remark this answers is not in the document]
Comment (hayakawa16) [reply src/Vts.Test/MonteCarlo/Detectors/DAWLayersDetectorsTests.cs:39]: Sounds good.

File: src/Vts.Test/MonteCarlo/Detectors/DAWLayersDetectorsTests.cs [hunk 2]
```diff
@@ -1232,35 +1229,35 @@ public void validate_DAW_TransmittedMTOfRhoAndSubregionHist()
         }
         // Reflected Momentum Transfer of X, Y and SubRegion
         [Test]
-        public void validate_DAW_ReflectedMTOfXAndYAndSubregionHist()
+        public void Validate_DAW_ReflectedMTOfXAndYAndSubregionHist()
```
[the remark this answers is not in the document]
Comment (hayakawa16) [reply src/Vts.Test/MonteCarlo/Detectors/DAWLayersDetectorsTests.cs:1232]: I must has missed this one.  I renamed all other unit tests I think.

**Review by lmalenfant — APPROVED** [review]

**Comment by lmalenfant** [comment]
I approved this PR so we can merge as soon as you would like, hopefully it will pass the quality gate.

**Comment by lmalenfant** [comment]
Let's merge this one and keep our fingers crossed that it passes code quality :)

**Pull request merged**